SwiftWisdom/Core/StandardLibrary/Numbers/Mathable.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  Mathable.swift
+//  SwiftWisdom
+//
+//  Created by Paul Rolfe on 2/12/16.
+//  Copyright © 2016 Intrepid. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public protocol Mathable: Comparable {
+    func + (lhs: Self, rhs: Self) -> Self
+    func / (lhs: Self, rhs: Self) -> Self
+    func * (lhs: Self, rhs: Self) -> Self
+    func - (lhs: Self, rhs: Self) -> Self
+    init(_ int: IntMax)
+}
+
+extension CollectionType where Generator.Element: Mathable {
+    
+    public func ip_average() -> Generator.Element? {
+        return ip_mean()
+    }
+    
+    public func ip_mean() -> Generator.Element? {
+        let zero = Generator.Element(0.toIntMax())
```

Comment:
We use zero a lot, it might be worth writing an extension on mathable:

```Swift
extension Mathable {
    public static var zero: Self {
        return self.init(0.toIntMax())
    }
}
```

Then here, it'd be:

```Swift
let zero = Generator.Element.zero
```